### PR TITLE
zephyr: micp & aics: Add BTP event

### PR DESCRIPTION
**Pull request by piotrnarajowski**
Enchance BTP events functionality for MICP and AICS

### CODE REVIEW

**Comment by piotrnarajowski** [comment]
@mkasenberg

**Comment by piotrnarajowski** [comment]
@sjanc

File: autopts/wid/micp.py [hunk 1]
```diff
@@ -113,10 +119,12 @@ def hdl_wid_8(params: WIDParams):
...
-    ev = stack.aics.wait_aics_state_ev(addr_type, addr, 30)
-    # TODO: Should return False if ev is None
-
-    return True
+    ev = stack.aics.wait_aics_procedure_ev(addr_type, addr, 10)
+    if ev[3] == 4:
+        # opcode == 0x04: unmute
+        return True
+    else:
+        return False
```
Comment (mkasenberg) [inline autopts/wid/micp.py:127]:
```suggestion
return False
```

File: autopts/wid/micp.py [hunk 2]
```diff
@@ -127,10 +135,12 @@ def hdl_wid_9(params: WIDParams):
...
-    ev = stack.aics.wait_aics_state_ev(addr_type, addr, 30)
-    # TODO: Should return False if ev is None
-
-    return True
+    ev = stack.aics.wait_aics_procedure_ev(addr_type, addr, 10)
+    if ev[3] == 5:
+        # opcode == 0x05: set manual gain
+        return True
+    else:
+        return False
```
Comment (mkasenberg) [inline autopts/wid/micp.py:143]:
```suggestion
return False
```

File: autopts/wid/micp.py [hunk 3]
```diff
@@ -141,10 +151,12 @@ def hdl_wid_10(params: WIDParams):
...
-    ev = stack.aics.wait_aics_state_ev(addr_type, addr, 30)
-    # TODO: Should return False if ev is None
-
-    return True
+    ev = stack.aics.wait_aics_procedure_ev(addr_type, addr, 30)
+    if ev[3] == 6:
+        # opcode == 0x06: set automatic gain
+        return True
+    else:
+        return False
```
Comment (mkasenberg) [inline autopts/wid/micp.py:159]:
```suggestion
return False
```

File: autopts/wid/micp.py [hunk 4]
```diff
@@ -254,40 +266,64 @@ def hdl_wid_20110(params: WIDParams):
...
-        # Performing opposite operation before the one that is demanded is
-        # necessary because PTS is already set to a value that is required
-        # in test case description.
+        if params.test_case_name == "MICP/CL/SPE/BI-02-C":
+            ev = stack.aics.wait_aics_procedure_ev(addr_type, addr, 10, remove=False)
+            if ev[3] == 2:
+                # opcode == 0x02
+                return True
+            else:
+                return False
```
Comment (mkasenberg) [inline autopts/wid/micp.py:281]:
```suggestion
            return False
```

File: autopts/wid/micp.py [hunk 5]
```diff
@@ -254,40 +266,64 @@ def hdl_wid_20110(params: WIDParams):
...
     elif "Mute" in params.description:
         btp.aics_mute(addr_type, addr)
+        if params.test_case_name == "MICP/CL/SPE/BI-04-C":
+            ev = stack.aics.wait_aics_procedure_ev(addr_type, addr, 10, remove=False)
+            if ev[3] == 3:
+                # opcode == 0x03
+                return True
+            else:
+                return False
```
Comment (mkasenberg) [inline autopts/wid/micp.py:290]:
```suggestion
            return False
```

File: autopts/wid/micp.py [hunk 6]
```diff
@@ -254,40 +266,64 @@ def hdl_wid_20110(params: WIDParams):
...
+    elif "Unmute" in params.description:
+        btp.aics_unmute(addr_type, addr)
+        if params.test_case_name == "MICP/CL/SPE/BI-03-C":
+            ev = stack.aics.wait_aics_procedure_ev(addr_type, addr, 10, remove=False)
+            if ev[3] == 4:
+                # opcode == 0x04
+                return True
+            else:
+                return False
```
Comment (mkasenberg) [inline autopts/wid/micp.py:299]:
```suggestion
            return False
```

File: autopts/wid/micp.py [hunk 7]
```diff
@@ -254,40 +266,64 @@ def hdl_wid_20110(params: WIDParams):
...
-        # Performing opposite operation before the one that is demanded is
-        # necessary because PTS is already set to a value that is required
-        # in test case description.
+        if params.test_case_name == "MICP/CL/SPE/BI-05-C":
+            ev = stack.aics.wait_aics_procedure_ev(addr_type, addr, 10, remove=False)
+            if ev[3] == 5:
+                # opcode == 0x05
+                return True
+            else:
+                return False
```
Comment (mkasenberg) [inline autopts/wid/micp.py:308]:
```suggestion
            return False
```

File: autopts/wid/micp.py [hunk 8]
```diff
@@ -254,40 +266,64 @@ def hdl_wid_20110(params: WIDParams):
...
     elif "Set Automatic" in params.description:
         btp.aics_auto_gain_set(addr_type, addr)
+        if params.test_case_name == "MICP/CL/SPE/BI-06-C":
+            ev = stack.aics.wait_aics_procedure_ev(addr_type, addr, 10, remove=False)
+            if ev[3] == 6:
+                # opcode == 0x06
+                return True
+            else:
+                return False
```
Comment (mkasenberg) [inline autopts/wid/micp.py:317]:
```suggestion
            return False
```

File: autopts/wid/micp.py [hunk 9]
```diff
@@ -254,40 +266,64 @@ def hdl_wid_20110(params: WIDParams):
...
     elif "Op Code: <WildCard: Exists>" in params.description:
         btp.aics_auto_gain_set(addr_type, addr)
+        if params.test_case_name == "MICP/CL/SPE/BI-08-C":
+            ev = stack.aics.wait_aics_procedure_ev(addr_type, addr, 10, remove=False)
+            if ev[3] == 6:
+                # opcode == 0x06
+                return True
+            else:
+                return False
```
Comment (mkasenberg) [inline autopts/wid/micp.py:326]:
```suggestion
            return False
```

**Review by mkasenberg — CHANGES_REQUESTED** [review]

**Review by mkasenberg — APPROVED** [review]

**Pull request merged**